openmdao/core/group.py

```diff
@@ -762,14 +789,16 @@ def _setup_var_data(self):
                 allprocs_abs2prom[type_] = {}
                 allprocs_prom2abs_list[type_] = OrderedDict()
 
-            group_inputs = []
-            for (myproc_abs_names, myproc_abs_names_discrete, myproc_discrete,
-                 myproc_prom2abs_list, all_abs2prom, myproc_abs2meta, oscale,
-                 rscale, ginputs) in gathered:
+            myrank = self.comm.rank
+            for rank, (myproc_abs_names, myproc_abs_names_discrete, myproc_discrete,
+                       myproc_prom2abs_list, all_abs2prom, myproc_abs2meta, oscale,
+                       rscale, ginputs) in enumerate(gathered):
                 self._has_output_scaling |= oscale
                 self._has_resid_scaling |= rscale
 
-                group_inputs.extend(ginputs)
+                if rank != myrank:
+                    for p, mlist in ginputs.items():
+                        self._group_inputs[p].extend(mlist)
```

Comment:
Is this a bug fix?

Reply:
The data structure was changed to make error checking easier later.